keyring.go

```diff
@@ -153,11 +153,15 @@ func (p *PrivateKeyRingFolder) Latest(id IssuerIdentifier) (*gabikeys.PrivateKey
 }
 
 func (p *PrivateKeyRingFolder) Iterate(id IssuerIdentifier, f func(sk *gabikeys.PrivateKey) error) error {
-	files, err := filepath.Glob(filepath.Join(p.path, fmt.Sprintf("%s*", id.String())))
+	files, err := filepath.Glob(filepath.Join(p.path, fmt.Sprintf("%s.xml", id.String())))
```

Comment:
> This is always exactly one file right? Can we not simply to an `os.Stat` to check whether the file exists? That makes more explicit what you want do do.

I checked: this is exactly what `filepath.Glob` does if there are no special characters like `*` in the filename, so it comes down to the same thing. However, I agree with you that it is better to make the code show more explicitly what is happening, so I changed it.

>  If I'm right you want to use `x.y.xml` if present and otherwise you want to look for `x.y.0.xml`, etc.

Not exactly: in this function we want to iterate over all files present in the folder, as opposed to looking for x.y.0.xml only when x.y.xml is not present. (The behaviour of this function in this regard is the same in the previous and current version of this PR.)

>  On top of that, I think the `.` is now interpreted as a regex `.` for one random character.

No, this is not a regex; instead it seems to follow the syntax of [shell patterns](https://www.gnu.org/software/findutils/manual/html_node/find_html/Shell-Pattern-Matching.html), in which the `.` is not a special character. More explicitly, the [godoc of `filepath.Glob`](https://pkg.go.dev/path/filepath#Glob) says it uses the syntax of [`filepath.Match`, whose godoc](https://pkg.go.dev/path/filepath#Match) does not mention `.`.

Reply:
Ah oke, I thought it was a normal regex, but indeed you're right. The code looks clear to me now. I forgot backwards compatibility reasons, so then indeed it is better to not enforce the indexes in the file name when giving multiple private keys.